### PR TITLE
Refactor Dataframe-related classes

### PR DESCRIPTION
**Pull request by VictorVelarde**
This PR includes several refactors:
- [x] Refactor inside `Dataframe` class into smaller, more meaningful functions (and removing some duplications)
- [x] New dataframe namespace, for Dataframe-related classes
- [x] Extract new `ViewportFeature` class generator from `Dataframe`
- [x] Extract new 'helper classes' from `Dataframe`:
  - [x] `FeatureIdsHelper`
  - [x] `ProjectionHelper`
  - [x] `ViewportHelper`

### CODE REVIEW

**Comment by VictorVelarde** [comment]
Hey @Jesus89 would you mind having a look at this one?

**Comment by Jesus89** [comment]
Sure!

File: src/renderer/dataframe/Dataframe.js [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+     * This method is very intensive in terms of computations, loading potentially a lot of CPU data to the GPU
+     *
+     */
+    bindRenderer (renderer) {
+        this.renderer = renderer;
+        this.height = this.getSize().height;
+        this._initHelpers();
+
+        // Load alphanumeric properties to WebGL textures
+        this.addProperties();
```
[the remark this answers is not in the document]
Comment (VictorVelarde) [reply src/renderer/dataframe/Dataframe.js]: Yes indeed. Thx!

[the remark this answers is not in the document]
Comment (Jesus89) [reply src/renderer/dataframe/Dataframe.js]: True that

**Comment by VictorVelarde** [comment]
Are you ok with merging this @elenatorro ?

File: src/renderer/dataframe/Dataframe.js [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+     *   - the textures for each property
+     *   - the buffers for vertices & normals
+     *   - the auxiliary textures for style properties
+     *
+     * This method is very intensive in terms of computations, loading potentially a lot of CPU data to the GPU
+     *
+     */
+    bindRenderer (renderer) {
+        this.renderer = renderer;
+        this.height = this.getSize().height;
```
Comment (elenatorro) [inline src/renderer/dataframe/Dataframe.js]: Maybe we can remove this line, and also the other assignment to update the dataframe height? I _think_ it's no longer needed anywhere

Comment (VictorVelarde) [reply src/renderer/dataframe/Dataframe.js]: I think I have found an external use. See next commit for a little refactor to remove it

**Pull request merged**